external-crates/move/crates/move-compiler/src/command_line/compiler.rs

```diff
@@ -96,21 +114,44 @@ enum PassResult {
     Typing(typing::ast::Program),
     HLIR(hlir::ast::Program),
     CFGIR(cfgir::ast::Program),
-    Compilation(Vec<AnnotatedCompiledUnit>, /* warnings */ Diagnostics),
+    Compilation(
+        Vec<AnnotatedCompiledUnit>,
+        Vec<ModuleIdent>,
+        /* warnings */ Diagnostics,
+    ),
 }
 
+pub type CompiledModuleInfoMap = BTreeMap<ModuleIdent, Arc<CompiledModuleInfo>>;
+
 #[derive(Clone)]
-pub struct FullyCompiledProgram {
-    pub files: MappedFiles,
-    pub parser: parser::ast::Program,
-    pub expansion: expansion::ast::Program,
-    pub naming: naming::ast::Program,
-    pub typing: typing::ast::Program,
-    pub hlir: hlir::ast::Program,
-    pub cfgir: cfgir::ast::Program,
-    pub compiled: Vec<AnnotatedCompiledUnit>,
+pub struct CompiledModuleInfo {
+    pub file_name: Symbol,
+    pub file_content: Arc<str>,
+    /// to compute address conflicts in expansion/translate.rs
+    /// (address map "inlined" into the module so that
+    /// modules are separated)
+    pub named_address_map: NamedAddressMap,
+    /// to create context in expansion/translate.rs
+    pub member_kinds: BTreeMap<Name, ModuleMemberKind>,
+    /// to build context in naming/translate.rs
+    pub resolved_members: BTreeMap<Symbol, ResolvedModuleMember>,
+    /// to extract macros in typing/translate.rs (need function bodies for this)
+    pub macro_infos: Option<(UseFuns, UniqueMap<FunctionName, Function>)>,
+    /// to construct program infos in shared/program_info.rs
+    /// (need both because typing info drops some info present at naming,
+    /// such as abilities, and naming info does not have all the types resolved)
+    pub naming_info: ModuleInfo,
+    pub typing_info: ModuleInfo,
+    /// to process private transfers in sui_mode/info.rs
+    pub private_transfers: BTreeMap<DatatypeName, TransferKind>,
+    /// to process modules in to_bytecode/translate.rs
+    pub dependency_order: usize,
```

Comment:
This information is present after typing, we could just add it to TypingInfo?

Reply:
Sounds like a good idea to me